filters/ema.hpp

```diff
@@ -1,109 +1,156 @@
-#ifndef SERENITY_EXPONENTIAL_MOVING_AVERAGE_HPP
-#define SERENITY_EXPONENTIAL_MOVING_AVERAGE_HPP
+#ifndef SERENITY_EXPONENTIAL_MOVING_AVERAGE_FILTER_HPP
+#define SERENITY_EXPONENTIAL_MOVING_AVERAGE_FILTER_HPP
 
 #include <math.h>
 
+#include <bits/unique_ptr.h>
```

Comment:
This is an internal header file, included by other library headers. Do not attempt to use it directly.

Use <memory>